### DISCUSSION TITLE
"position" returning null in real-time race updates

### DISCUSSION THREAD

**Discussion by fredericomattos** [br-g/openf1#24] [Q&A]
When collecting real-time race data:

```
https://api.openf1.org/v1/position?meeting_key=latest&session_key=latest
```

I noticed that there are returns with the `position` field as `null`:

![image](https://github.com/br-g/openf1/assets/58320500/3c86dce3-13b5-4a6b-947f-fb05813004a1)

So that I can keep the position updated with the most recent record, I sort by the most recent date, filter those that do not return `null`, remove duplicates so as not to keep older records and finally sort by `position` like this:

```python
from time import sleep
import pandas as pd

while True:
    df_race = pd.read_json("https://api.openf1.org/v1/position?meeting_key=latest&session_key=latest")
    df_drivers = pd.read_json("https://api.openf1.org/v1/drivers?session_key=latest")
    df_merged = pd.merge(df_race, df_drivers[['driver_number', 'broadcast_name']], on='driver_number')
    df_merged = df_merged.sort_values(by=["date"], ascending=False)
    df_merged = df_merged[~df_merged["position"].isna()]
    df_merged = df_merged.drop_duplicates(subset=["driver_number"])
    df_merged["position"] = df_merged["position"].astype(int)
    df_merged = df_merged.sort_values(by=["position"], ascending=True)
    df_merged.to_csv("formula_1.csv", index=False)
    sleep(1)
```
I made a video demonstrating that it took 1 minute and 12 seconds for Zhou's position to go from 11th to 17th (the moment the position changed from 16th to 17th I started recording the video):

https://github.com/br-g/openf1/assets/58320500/c85a37a9-bf54-4806-9ffc-18ecde6fb79b

I would like to understand what I am doing wrong, because in the documentation it says that updates are provided "3 seconds after a live event", so it is definitely an error on my part but due to lack of knowledge I was unable to understand my difficulty in keeping updated In real time.

**Comment by fredericomattos** [comment]
Attached video that was broken has been updated successfully ✅

**Comment by fredericomattos** [comment]
Add info: I noticed that in Practice 1 and Practice 2 in Saudi Arabia everyone delivered the "position" value perfectly, so it seems to me that the aforementioned case happens specifically during the official race..

**Comment by br-g** [comment]
Thanks for reporting this issue!
I will have a closer look when I find a bit of time
